src/main/java/assets/game.js

```diff
@@ -42,11 +45,22 @@ function markHits(board, elementId, surrenderText) {
             } else if (sonar.result === "SHOWSHIP") {
                 className = "occupied";
             }
-
             document.getElementById(elementId).rows[sonar.location.row - 1].cells[sonar.location.column.charCodeAt(0) - 'A'.charCodeAt(0)].classList.add(className);
         });
     }
 
+    if(board.spaceL){
+        board.spaceL.forEach((space) => {
+            let className;
+            if(space.result == "SHOWNOSHIP"){
+                className = "empty";
+            }else if (space.result == "SHOWSHIP"){
+                className = "occupied";
+            }
+        document.getElementById(elementId).rows[space.location.row - 1].cells[space.location.column.charCodeAt(0) - 'A'.charCodeAt(0)].classList.add(className);
+        });
+    }
+
```

Comment:
Great! This method leads to some code repetition, but in this case that is hard to avoid, so this is a good middle ground.